High contrast mode support

cee-chen opened the pull request:
## Summary

closes https://github.com/elastic/eui/issues/8133. See also https://github.com/elastic/eui/issues/7509#issuecomment-2260692852

### Screenshots

<img width="1484" alt="" src="https://github.com/user-attachments/assets/76d35c25-9af7-4555-871f-2b9e1cfa1ddd">

<img width="430" alt="" src="https://github.com/user-attachments/assets/c1ade711-a55b-4961-ab51-61039a5e8810"> <img width="300" alt="" src="https://github.com/user-attachments/assets/828a6b3f-bb8f-42f2-ba05-0a4e91bbf054">

| Popover | Tooltips |
|--------|--------|
| <img width="200" alt="" src="https://github.com/user-attachments/assets/44dbdb21-9483-49a3-b7c9-fd9c84d1590e"> <img width="250" alt="" src="https://github.com/user-attachments/assets/4992d8f2-df0b-40c1-8121-07403e5ebfee"> <img width="300" alt="" src="https://github.com/user-attachments/assets/7309cb0b-ee03-41bc-b32e-ae7faa6ed96b"> <img width="300" alt="" src="https://github.com/user-attachments/assets/87baca23-14db-4d24-942b-0dd5aed664f4"> | <img width="314" alt="" src="https://github.com/user-attachments/assets/ac543c61-8564-4ddf-b1c7-4983fc3ad22b"> <img width="206" alt="" src="https://github.com/user-attachments/assets/b262e633-c3ef-4b2f-8f2f-11c3ef722f46"> <br>Windows high contrast:<br> <img width="289" alt="" src="https://github.com/user-attachments/assets/b5bc7cc3-16f5-4fcd-bbbc-f43896f919cd"> <img width="159" alt="" src="https://github.com/user-attachments/assets/701f3bb3-02c0-43e4-a70d-179092ddfd34">  |

### Related PRs

> [!note]
> This is a feature branch. This means that it contains a massive changeset that has been reviewed in atomic chunks. Please see the below individual PRs for discussions, changes, and reviews:

- #8142
- #8174
- #8193 
- #8199
- #8200
- #8205 
- #8210
- #8215
- #8218
- #8219 
- #8223
- #8225
- #8226
- #8234

### Extra reading/reference:

-  https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast
- https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors
- https://kilianvalkhof.com/2023/css-html/i-no-longer-understand-prefers-contrast/

## QA

- Generally smoke test [the docs](https://eui.elastic.co/pr_8232/#/theming/high-contrast-mode) on both Mac and Windows high contrast modes to check for any visual issues

### General checklist

- Browser QA
    - [x] Checked in both **light and dark** modes
    - [x] Checked in both [MacOS](https://support.apple.com/lv-lv/guide/mac-help/unac089/mac) and [Windows](https://support.microsoft.com/en-us/windows/turn-high-contrast-mode-on-or-off-in-windows-909e9d89-a0f9-a3a9-b993-7a6dcee85025) **high contrast modes**
    - [x] Checked in **mobile**
    - [x] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
    - [x] Checked for **accessibility** including keyboard-only and screenreader modes
- Docs site QA
    - [x] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**
    - [x] Props have proper **autodocs** (using `@default` if default values are missing) and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/playgrounds.md)**
    ~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
- Code quality checklist
    - [x] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/unit-testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/cypress-testing.md) tests**
    - [x] Updated **[visual regression tests](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/visual-regression-testing.md)**
- Release checklist
    - [x] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md)** entry exists and is marked appropriately.
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A - no designs for high contrast mode (currently)

cee-chen committed:
Remove `no-extra-semicolons` stylelint rule

- at this point it's creating more false positives with nested template literals than it is being helpful

cee-chen committed:
[EuiProvider][EuiThemeProvider] `highContrastMode` set up and documentation (#8142)

cee-chen committed:
[High Contrast Mode] Panels, modals, flyouts, toasts, popovers, and tooltips (#8174)

cee-chen committed:
[High Contrast Mode] Form controls (#8193)

cee-chen committed:
[High Contrast Mode] Allow forced high contrast themes to also force color mode (#8200)

cee-chen committed:
[High Contrast Mode] Ranges, color pickers, and date pickers (#8199)

cee-chen committed:
[High contrast mode] DRY utilities (#8205)

cee-chen committed:
[High Contrast Mode] Buttons, datepicker redux, code blocks, and avatars (#8210)

cee-chen committed:
[High Contrast Mode] Horizontal rules, side navs, tabs, and breadcrumbs (#8218)

cee-chen committed:
[High Contrast Mode] Loading components and beacons (#8215)

cee-chen committed:
[High Contrast Mode] Markdown, highlighting, misc text fixes (#8219)

cee-chen committed:
[High Contrast Mode] EuiKeyPadMenu, EuiImage, EuiOverlayMask, EuiBottomBar, EuiToken (#8225)

cee-chen committed:
[High Contrast Mode] Steps, timelines, and comment lists (#8223)

cee-chen committed:
[High Contrast Mode] Tables and data grid (#8226)

cee-chen committed:
Update changelog file

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8232/
- Storybook: https://eui.elastic.co/pr_8232/storybook
- New docs: https://eui.elastic.co/pr_8232/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237)
* Commit: 38ddf2e65c76c131ef426ebc78d209eb19eeae3b

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237","number":3237,"commit":"38ddf2e65c76c131ef426ebc78d209eb19eeae3b"}],"number":3237}
buildkite-pr-comment-->

cee-chen commented:
I'll push up any minor visual tweaks/bugfixes directly to this PR. Already noticing one or two in the final smoke test 😅

mgadewoll commented on packages/eui/changelogs/upcoming/8232.md:
```diff
@@ -0,0 +1,4 @@
+- Updated `EuiProvider` `and `EuiThemeProvider` with a new `highContrastMode`
```
Typo: there is a `'` too much before the `and`

cee-chen replied:
Derp, thanks Lene!
```suggestion
- Updated `EuiProvider` and `EuiThemeProvider` with a new `highContrastMode`
```

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8232/
- Storybook: https://eui.elastic.co/pr_8232/storybook
- New docs: https://eui.elastic.co/pr_8232/new-docs

mgadewoll commented:
ℹ️ We won't merge this PR just yet as we first need to rebase and update Borealis with these changes. Otherwise the Borealis feature branch would be blocked from releases.

cee-chen committed:
[PR feedback] Changelog typo

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8232/
- Storybook: https://eui.elastic.co/pr_8232/storybook
- New docs: https://eui.elastic.co/pr_8232/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3241)
* Commit: b19de7b1ac944001644fad8da4c5fa391ae70207

### History
* :green_heart: [Build #3239](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3239) succeeded 50502ee73a02396b86d066716e44e3f4f8a8fe3c
* :green_heart: [Build #3237](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237) succeeded 38ddf2e65c76c131ef426ebc78d209eb19eeae3b

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3241","number":3241,"commit":"b19de7b1ac944001644fad8da4c5fa391ae70207"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3239","number":3239,"commit":"50502ee73a02396b86d066716e44e3f4f8a8fe3c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237","number":3237,"commit":"38ddf2e65c76c131ef426ebc78d209eb19eeae3b"}],"number":3241}
buildkite-pr-comment-->

mgadewoll committed:
docs: ensure high contrast mode works in EUI+

- enables workspace EUI version for docusaurus-theme

- fixes theme vs colorMode setting on EuiProvider in EUI+

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8232/
- Storybook: https://eui.elastic.co/pr_8232/storybook
- New docs: https://eui.elastic.co/pr_8232/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3242)
* Commit: 8d61392838594997b27f50ef8ec5924408a1228b

### History
* :green_heart: [Build #3241](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3241) succeeded b19de7b1ac944001644fad8da4c5fa391ae70207
* :green_heart: [Build #3239](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3239) succeeded 50502ee73a02396b86d066716e44e3f4f8a8fe3c
* :green_heart: [Build #3237](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237) succeeded 38ddf2e65c76c131ef426ebc78d209eb19eeae3b

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3242","number":3242,"commit":"8d61392838594997b27f50ef8ec5924408a1228b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3241","number":3241,"commit":"b19de7b1ac944001644fad8da4c5fa391ae70207"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3239","number":3239,"commit":"50502ee73a02396b86d066716e44e3f4f8a8fe3c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237","number":3237,"commit":"38ddf2e65c76c131ef426ebc78d209eb19eeae3b"}],"number":3242}
buildkite-pr-comment-->

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8232/
- Storybook: https://eui.elastic.co/pr_8232/storybook
- New docs: https://eui.elastic.co/pr_8232/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3243)
* Commit: 578de0bac3e1f4c6af82bf88738125edd41130cf

### History
* :green_heart: [Build #3242](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3242) succeeded 8d61392838594997b27f50ef8ec5924408a1228b
* :green_heart: [Build #3241](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3241) succeeded b19de7b1ac944001644fad8da4c5fa391ae70207
* :green_heart: [Build #3239](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3239) succeeded 50502ee73a02396b86d066716e44e3f4f8a8fe3c
* :green_heart: [Build #3237](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237) succeeded 38ddf2e65c76c131ef426ebc78d209eb19eeae3b

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3243","number":3243,"commit":"578de0bac3e1f4c6af82bf88738125edd41130cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3242","number":3242,"commit":"8d61392838594997b27f50ef8ec5924408a1228b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3241","number":3241,"commit":"b19de7b1ac944001644fad8da4c5fa391ae70207"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3239","number":3239,"commit":"50502ee73a02396b86d066716e44e3f4f8a8fe3c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237","number":3237,"commit":"38ddf2e65c76c131ef426ebc78d209eb19eeae3b"}],"number":3243}
buildkite-pr-comment-->

cee-chen commented:
I forced pushed to remove some of the smaller fixes I pre-emptively pushed up. I'm just going to open a new (final 💀) PR with various fixes and updates I found in my last review: https://github.com/elastic/eui/pull/8234

cee-chen committed:
[High Contrast Mode] new_final_final(2).psd (#8234)

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8232/
- Storybook: https://eui.elastic.co/pr_8232/storybook
- New docs: https://eui.elastic.co/pr_8232/new-docs

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3251)
* Commit: 34e5db8849ad8261ed5d67954710a09dd895e7e4

### Failed CI Steps
* [:hammer: EUI pull request test](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3251#0193c004-ccde-4253-93f9-3f7235876f8d)

### History
* :green_heart: [Build #3243](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3243) succeeded 578de0bac3e1f4c6af82bf88738125edd41130cf
* :green_heart: [Build #3242](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3242) succeeded 8d61392838594997b27f50ef8ec5924408a1228b
* :green_heart: [Build #3241](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3241) succeeded b19de7b1ac944001644fad8da4c5fa391ae70207
* :green_heart: [Build #3239](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3239) succeeded 50502ee73a02396b86d066716e44e3f4f8a8fe3c
* :green_heart: [Build #3237](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237) succeeded 38ddf2e65c76c131ef426ebc78d209eb19eeae3b

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3251","number":3251,"commit":"34e5db8849ad8261ed5d67954710a09dd895e7e4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3243","number":3243,"commit":"578de0bac3e1f4c6af82bf88738125edd41130cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3242","number":3242,"commit":"8d61392838594997b27f50ef8ec5924408a1228b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3241","number":3241,"commit":"b19de7b1ac944001644fad8da4c5fa391ae70207"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3239","number":3239,"commit":"50502ee73a02396b86d066716e44e3f4f8a8fe3c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3237","number":3237,"commit":"38ddf2e65c76c131ef426ebc78d209eb19eeae3b"}],"number":3251}
buildkite-pr-comment-->

mgadewoll commented:
ℹ️ @cee-chen We finally are getting back to this! ⏲️ The changes done your PR have been moved to [this new PR](https://github.com/elastic/eui/pull/8444) to align them with the latest state of `main` and required updates for the new theme. I'm closing this PR in favor of the new one.
